packages/charts/src/chart_types/xy_chart/utils/series.ts

```diff
@@ -299,7 +299,7 @@ const getSortedDataSeries = (
   xValues: Set<string | number>,
   xScaleType: ScaleType,
 ): DataSeries[] =>
-  dataSeries.map(({ data, ...rest }) => ({
+  [...dataSeries].map(({ data, ...rest }) => ({
     ...rest,
     data: data.sort(datumXSortPredicate(xScaleType, [...xValues.values()])),
   }));
```

Comment:
```suggestion
   dataSeries.map(({ data, ...rest }) => ({
    ...rest,
    data: [...data].sort(datumXSortPredicate(xScaleType, [...xValues.values()])),
  }));
```
It looks like we need to shallow copy the `data` object that you are sorting, not the `dataSeries` one that is not changed

Reply:
Thanks, what was I thinking with this? 🤣

Reply:
b22c647a1fce16c8e5f47ecaf003e64f6962a4f1